### PR TITLE
xds: implement XdsChannelCredentials

### PR DESCRIPTION
**Pull request by sanjaypujare**
@ejona86 PTAL.

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/XdsChannelCredentials.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.grpc.ChannelCredentials;
+import io.grpc.netty.InternalNettyChannelCredentials;
+import io.grpc.netty.InternalProtocolNegotiator;
+import io.grpc.xds.internal.sds.SdsProtocolNegotiators;
+
+public class XdsChannelCredentials {
```
Comment (ejona86) [inline xds/src/main/java/io/grpc/xds/XdsChannelCredentials.java:26]: ExperimentalApi?

**Review by ejona86 — COMMENTED** [review]
Seems fair. Didn't approve since it is draft.

File: xds/src/main/java/io/grpc/xds/XdsChannelCredentials.java (hunk 1 above)
Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/XdsChannelCredentials.java:28]: Done

**Comment by sanjaypujare** [comment]
> Seems fair. Didn't approve since it is draft.

Removed draft...

**Review by ejona86 — APPROVED** [review]

**Pull request merged**